internal/irc/service.go

```diff
@@ -127,7 +126,7 @@ func (s *service) startNetwork(network domain.IrcNetwork) error {
 	if existingHandler, found := s.handlers[network.ID]; found {
 		s.log.Debug().Msgf("starting network: %s", network.Name)
 
-		if !existingHandler.client.Connected() {
+		if existingHandler.Stopped() {
```

Comment:
From my reading of the intended behavior here, what really matters here (and below in this file) is not whether the handler is actively connected (has a working connection to the remote IRC server at this moment in time), but whether it was manually stopped.